test/functional/testlib.bash

```diff
@@ -2693,3 +2705,40 @@ assert_files_not_equal() {
 	fi
 
 }
+
+# Displays a menu of functions that can be used from within a test. Functions considered
+# "internal" are not displayed.
+# Functions are not included in the list by default, to mark a function to be included in
+# the help menu, add a comment immediatelly after the function opening bracket with either
+# the keyword swupd_function or assertion depending on the type of function.
+# Parameters:
+# - prefix: a prefix that could be used to filter functions
+testlib() {
+
+	local prefix=$1
+	local funcs
+	local assertions
+
+	funcs=$(grep "$prefix.*() { [#] swupd_function" "$FUNC_DIR"/testlib.bash | cut -f1 -d "(")
+	assertions=$(grep "$prefix.*() { [#] assertion" "$FUNC_DIR"/testlib.bash | cut -f1 -d "(")
```

Comment:
For these regex to match the function prefixes, they should start at the beginning of the line.

Reply:
You are right. But I prefer the way it is now.
So:

     $ testlib not

will print all functions with not in the name, like:

    assert_status_is_not
    assert_dir_not_exists
    assert_file_not_exists
    assert_not_in_output
    assert_is_not_output
    assert_regex_not_in_output
    assert_regex_is_not_output
    assert_not_equal
    assert_files_not_equal

Reply:
In that case, I think the prefix argument name and description should be changed to remove confusion.

Reply:
that is a good idea @jwakre, I'll update that